tests/local/teleporter/teleporter_suite_test.go

```diff
@@ -0,0 +1,188 @@
+package teleporter_test
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	teleporterFlows "github.com/ava-labs/teleporter/tests/flows/teleporter"
+	registryFlows "github.com/ava-labs/teleporter/tests/flows/teleporter/registry"
+	"github.com/ava-labs/teleporter/tests/local"
+	deploymentUtils "github.com/ava-labs/teleporter/utils/deployment-utils"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	teleporterByteCodeFile  = "./out/TeleporterMessenger.sol/TeleporterMessenger.json"
+	warpGenesisTemplateFile = "./tests/utils/warp-genesis-template.json"
+
+	teleporterMessengerLabel = "TeleporterMessenger"
+	upgradabilityLabel       = "upgradability"
+	utilsLabel               = "utils"
+	validatorSetSigLabel     = "ValidatorSetSig"
+	validatorManagerLabel    = "ValidatorManager"
+)
+
+var (
+	LocalNetworkInstance *local.LocalNetwork
+)
+
+func TestTeleporter(t *testing.T) {
+	if os.Getenv("RUN_E2E") == "" {
+		t.Skip("Environment variable RUN_E2E not set; skipping E2E tests")
+	}
+
+	RegisterFailHandler(ginkgo.Fail)
+	ginkgo.RunSpecs(t, "Teleporter e2e test")
+}
+
+// Define the Teleporter before and after suite functions.
+var _ = ginkgo.BeforeSuite(func() {
```

Comment:
Is this setup the same for all the suites? Can we extract this function out somewhere common?

Reply:
The only difference is that `governance` and `validator-manager` suites omit setting up Teleporter. I'm open to suggestions, but IMO the set of network setup functions are sufficiently modular to cover our test cases.

Reply:
Sorry - got my wires crossed. This [WIP branch](https://github.com/ava-labs/teleporter/compare/e2e-refactor-separate-suites...e2e-refactor-cleanup?expand=1) makes the change I described. With it, the `governance` and `validator-manager` suites have no dependency on the Teleporter contracts